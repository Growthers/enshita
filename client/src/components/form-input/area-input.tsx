import { forwardRef } from "react";
import styles from "~/styles/components/form-input/input.module.scss";
import type { AreaInputProperties } from "./type/model";

// eslint-disable-next-line react/display-name
const AreaInput = forwardRef<HTMLTextAreaElement, AreaInputProperties>(
  ({ inputStyles, error, ...rest }, ref) => (
    <>
      <textarea ref={ref} className={inputStyles} {...rest} />
      {error && (
        <p aria-label="error message" className={styles["enshita-input-error"]}>
          {error.message}
        </p>
      )}
    </>
  ),
);

export { AreaInput };
