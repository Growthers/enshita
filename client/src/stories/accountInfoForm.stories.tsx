// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
