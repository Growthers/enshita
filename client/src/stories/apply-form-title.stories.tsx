import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
