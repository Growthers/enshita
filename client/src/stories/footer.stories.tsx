// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
