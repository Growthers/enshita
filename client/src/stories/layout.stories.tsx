import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
