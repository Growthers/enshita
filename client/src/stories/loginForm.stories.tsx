// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
