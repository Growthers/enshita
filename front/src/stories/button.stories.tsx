import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
