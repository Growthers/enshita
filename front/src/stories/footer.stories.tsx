import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
