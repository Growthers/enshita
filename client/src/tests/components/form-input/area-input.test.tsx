import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/form-input/area-input.stories";

const { Default, Error, InputFieldFilled } = composeStories(stories);

describe("(components) text area input", () => {
  test("label should be test", () => {
    const { getByLabelText } = render(<Default />);
    expect(getByLabelText("test")).toBeInTheDocument();
  });
  test("input is correctly entered", async () => {
    const { container, getByLabelText } = render(<InputFieldFilled />);
    await InputFieldFilled.play({ canvasElement: container });
    const input = getByLabelText("test") as HTMLInputElement;
    expect(input.value).toEqual("This is test.");
  });
  test("error message is exist", () => {
    const { getByLabelText } = render(<Error />);
    expect(getByLabelText("error message")).toBeInTheDocument();
  });
  test("(normal) take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(error) take snapshot", () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });
});
