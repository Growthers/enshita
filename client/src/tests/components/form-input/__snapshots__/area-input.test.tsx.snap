// Vitest Snapshot v1

exports[`(components) text area input > (error) take snapshot 1`] = `
<div>
  <textarea
    aria-label="test"
    id="test"
  />
  <p
    aria-label="error message"
    class="_enshita-input-error_k8dqh_1"
  >
    1000字以内で入力してください
  </p>
</div>
`;

exports[`(components) text area input > (normal) take snapshot 1`] = `
<div>
  <textarea
    aria-label="test"
    id="test"
  />
</div>
`;
